pkg/reconciler/route/route.go

```diff
@@ -144,18 +145,18 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 	// and may not have had all of the assumed defaults specified.  This won't result
 	// in this getting written back to the API Server, but lets downstream logic make
 	// assumptions about defaulting.
-	r.SetDefaults(ctx)
+	r.SetDefaults(v1beta1.WithUpgradeViaDefaulting(ctx))
 	r.Status.InitializeConditions()
 
-	// There are no conditions that would trigger this, but if they were we'd have a
-	// block like this here (as the other controllers).
-	// if err := r.ConvertUp(ctx, &v1beta1.Route{}); err != nil {
-	// 	if ce, ok := err.(*v1alpha1.CannotConvertError); ok {
-	// 		r.Status.MarkResourceNotConvertible(ce)
-	// 	} else {
-	// 		return err
-	// 	}
-	// }
+	if err := r.ConvertUp(ctx, &v1beta1.Route{}); err != nil {
+		// There are no conditions that would trigger this, but if they were we'd have a
+		// block like this here (as the other controllers).
+		// if ce, ok := err.(*v1alpha1.CannotConvertError); ok {
+		// 	r.Status.MarkResourceNotConvertible(ce)
+		// } else {
+		return err
```

Comment:
The real `return err` here among the commented out code is confusing.

Reply:
I removed the surrounding comment